test/cpp/client/credentials_test.cc

```diff
@@ -419,8 +423,89 @@ TEST(CredentialsTest, TlsChannelCredentialsWithCrlProviderAndDirectory) {
   GPR_ASSERT(channel_credentials.get() != nullptr);
 }
 
-// TODO(gtcooke94) - Add test to make sure Tls*CredentialsOptions does not leak
-// when not moved into TlsCredentials
+TEST(CredentialsTest, TlsCredentialsOptionsDoesNotLeak) {
+  // TlsCredentialsOptions does not leak.
```

Comment:
What do you think about separating each of these into individual test cases?

I prefer it - imagine this starts failing sometime, it's immediately somewhat annoying to start to debug which of the 5 tests in this one test is failing.

Reply:
Done.